mods/default/torch.lua

```diff
@@ -3,152 +3,116 @@
 -- support for MT game translation.
 local S = default.get_translator
 
-local function on_flood(pos, oldnode, newnode)
-	minetest.add_item(pos, ItemStack("default:torch 1"))
-	-- Play flame-extinguish sound if liquid is not an 'igniter'
-	local nodedef = minetest.registered_items[newnode.name]
-	if not (nodedef and nodedef.groups and
-			nodedef.groups.igniter and nodedef.groups.igniter > 0) then
-		minetest.sound_play(
-			"default_cool_lava",
-			{pos = pos, max_hear_distance = 16, gain = 0.1},
-			true
-		)
+function default.register_torch(name, def)
+	def.drawtype = "mesh"
```

Comment:
def should be copied with `table.copy` before making changes

Reply:
What does that make sense?

Reply:
You don't want to pollute the caller's `def` table with changes, e.g. if it is reused.